CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntaxbuilder/BuildableNodesFile.swift

```diff
@@ -103,7 +103,8 @@ private func createConvenienceInitializer(node: Node) -> InitializerDeclSyntax?
 
   return InitializerDeclSyntax(
     leadingTrivia: .docLineComment("/// A convenience initializer that allows initializing syntax collections using result builders") + .newline,
-    modifiers: [DeclModifierSyntax(name: .keyword(.public))],
+    attributes: [.attribute(AttributeSyntax("@_disfavoredOverload"))],
```

Comment:
Without this, we get a `Ambiguous use of 'init'`

Reply:
I just tried it and we only need this if the built type is optional. Only in that case is the parameter is defaulted in both the builder initializer and the normal initializer in SwiftSyntax.

Since these convenience initializers only exist for the result builders now, I think we should not default their value, which also resolves the ambiguity. I just tried that, it requires a few updates to the test cases but IMO they all provide more clarity.